assets/src/js/Components/ArchivePicker/ArchivePicker.js

```diff
@@ -356,6 +364,23 @@ export default function ArchivePicker({view = ADMIN_VIEW}) {
     });
   };
 
+  const setCustomFeaturedImage = async imgId => {
+    await wp.data.dispatch('core/editor').editPost({featured_media: imgId});
+    dispatch({type: ACTIONS.CLOSE_SIDEBAR});
+    document.querySelector('.media-modal-close .media-modal-icon').click();
+  };
+
+  // Prevent multiple instances of the Media Upload Modal.
+  const mediaModalPopup = document.querySelectorAll('.media-modal');
```

Comment:
I'd try to move the `mediaModalPopup` as a state hook, and declare it within the initial use effect hook. Then you'll convert the code more Reactive. Also will avoid to re-deeclare in every render.

For example
```
const mediaModalPopup = document.querySelectorAll('.media-modal');
```
Replace with
```
useEffect(() => {
if(document.querySelectorAll('.media-modal')) {
   setMediaModalPopup(document.querySelectorAll('.media-modal'));
}, []);
```

Reply:
What do you think?